tests/cli/test_build.py

```diff
@@ -116,18 +116,6 @@ def test_raises_if_gen_config_target_not_passed_when_required(
             self.assertIsNotNone(result.exception)
             self.assertRegex(result.output, "--mbed-target")
 
-    def test_raises_if_gen_config_target_toolchain_not_passed(
```

Comment:
Does click not raise an exception if the target and toolchain aren't passed? What should we check for instead? What tests do we have that would make sure that if we remove "required=True" on accident, we still ensure each arg is required?

Reply:
As per commit message:
```
Note: The test case where both target and toolchain options are
missing has been removed, because the "click" only checks one argument
at a time. The existing tests for one option missing (one for
toolchain, one for target) are sufficient.
```
I think two tests above the removed one already serve the purpose.

Reply:
So previously we had three tests
* target not passed
* toolchain not passed
* target & toolchain not passed

Because click checks only one argument at a time, only the first two make sense now. But they are still sufficient to ensure users pass both target & toolchain, to not have any problems.

Reply:
OK, I didn't see we were testing them individually already and expected we had to split them.